raster/src/main/scala/geotrellis/raster/ArrayMultibandTile.scala

```diff
@@ -428,4 +428,16 @@ class ArrayMultibandTile(_bands: Array[Tile]) extends MultibandTile with MacroMu
 
     new ArrayMultibandTile(newBands)
   }
+
+  override def equals(other: Any): Boolean = other match {
+    case that : ArrayMultibandTile =>
+      var result = (bandCount == that.bandCount)
```

Comment:
There was a null check here before, was that so that `that.bandCount` would not throw? That seems reasonable.

Also for types we don't put a right space, so `case that: ArrayMultibandTile`. Nitpicking :)

I propose

```scala
override def equals(other: Any): Boolean =
  other match {
    case that: ArrayMultibandTile if that != null =>
      var result = (bandCount == that.bandCount)
      
      cfor(0)(result && _ < bandCount, _ + 1) { i =>
        if (band(i) != that.band(i)) result = false
      }

      result
    case _ =>
      false
  }
```